sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidPlanner.java

```diff
@@ -744,18 +755,28 @@ public T next()
 
     private SqlNode query;
 
-    private ParsedNodes(@Nullable SqlExplain explain, @Nullable SqlInsert insert, SqlNode query)
+    @Nullable
+    private String ingestionGranularity;
+
+    private ParsedNodes(
+        @Nullable SqlExplain explain,
+        @Nullable SqlInsert insert,
```

Comment:
Would it make sense to have this be a DruidSqlInsert, so the granularity doesn't need to be provided separately? I'm asking since my understanding is we'll _always_ get a DruidSqlInsert, even if the user doesn't provide PARTITION BY or CLUSTER BY.

Reply:
Yep, your understanding of the code is correct. That is we always hit our DruidSqlInsert production rule, irrespective of whether we have a simple INSERT or the one containing clauses. But just to ensure that I wasn't breaking anything that already existed while prototyping (and this overshadowing of the custom rule is not seen in the code), I am simply returning the SqlInsert statement in case both partition and cluster clauses are empty. 

Since replacing SqlInsert with DruidSqlInsert seems to be a better approach, I think we can remove separate handling of SqlInsert and DruidSqlInsert (and make the corresponding changes in the production rule)

Reply:
Update: We might not be able to do it cleanly due to SqlExplain. Explained more in the related comment.

Reply:
I see. But we'll need EXPLAIN on INSERT to work eventually, so we'll need to clean this up at some point.